spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-ai/src/main/java/com/alibaba/cloud/ai/tongyi/audio/transcription/TongYiAudioTranscriptionClient.java

```diff
@@ -0,0 +1,172 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.ai.tongyi.audio.transcription;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.alibaba.cloud.ai.tongyi.audio.ParaformerModels;
+import com.alibaba.cloud.ai.tongyi.audio.transcription.api.AudioTranscriptionPrompt;
+import com.alibaba.cloud.ai.tongyi.audio.transcription.api.AudioTranscriptionResponse;
+import com.alibaba.cloud.ai.tongyi.audio.transcription.api.TranscriptionClient;
+import com.alibaba.dashscope.audio.asr.transcription.Transcription;
+import com.alibaba.dashscope.audio.asr.transcription.TranscriptionParam;
+import com.alibaba.dashscope.audio.asr.transcription.TranscriptionQueryParam;
+import com.alibaba.dashscope.audio.asr.transcription.TranscriptionResult;
+import com.alibaba.dashscope.audio.asr.transcription.TranscriptionTaskResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.util.Assert;
+
+
+
+/**
+ * TongYiAudioTranscriptionClient is a client for TongYi audio transcription service for
+ * Spring Cloud Alibaba AI.
+ * @author: xYLiu
+ * @date: 2024/5/4
+ */
+
+public class TongYiAudioTranscriptionClient implements TranscriptionClient {
+
+	private final Logger logger = LoggerFactory.getLogger(getClass());
+
+	/**
+	 * TongYi models options.
+	 */
+	private final TongYiAudioTranscriptionOptions defaultOptions;
+
+	/**
+	 * TongYi models api.
+	 */
+	private final Transcription transcription;
+
+	public TongYiAudioTranscriptionClient(Transcription transcription) {
+		this(null, transcription);
+	}
+
+	public TongYiAudioTranscriptionClient(TongYiAudioTranscriptionOptions defaultOptions,
+			Transcription transcription) {
+		Assert.notNull(transcription, "transcription must not be null");
+		Assert.notNull(defaultOptions, "defaultOptions must not be null");
+
+		this.defaultOptions = defaultOptions;
+		this.transcription = transcription;
+	}
+
+	@Override
+	public AudioTranscriptionResponse call(AudioTranscriptionPrompt prompt) {
+		var param = merge(prompt.getOptions());
+		List<String> urls = prompt.getAudioUrl().getfileUrls();
+		TranscriptionParam transcriptionParam = toTranscriptionParam(param);
+		transcriptionParam.setFileUrls(urls);
+		logger.info(transcriptionParam.toString());
+
+		List<TranscriptionTaskResult> taskResultList = null;
+		try {
+			// 提交转写请求
+			TranscriptionResult result = transcription.asyncCall(transcriptionParam);
+			// 等待转写完成
+			result = transcription.wait(TranscriptionQueryParam
+					.FromTranscriptionParam(transcriptionParam, result.getTaskId()));
+			// 获取转写结果
+			taskResultList = result.getResults();
+			return new AudioTranscriptionResponse(taskResultList);
+		}
+		catch (Exception e) {
+			System.out.println("error: " + e);
```

Comment:
Should use `throw`, not `System.out.println()`

Reply:
reply done